naptime-testing/src/test/scala/org/coursera/naptime/NonNestedMacroTests.scala

```diff
@@ -246,16 +253,28 @@ class NonNestedMacroTests extends AssertionsForJUnit with MockitoSugar with Reso
   @Test
   def actionTest(): Unit = {
     val actionRequest = mkRequest(None, "POST",
-      query = Map("action" -> "batchModify", "someParam" -> "daphne~coursera.org"))
-    checkRouter(actionRequest, "batchModify")
-    verify(instance).batchModify(Some(ComplexEmailType("daphne", "coursera.org")))
+      query = Map("action" -> "actionWithParams", "someParam" -> "daphne~coursera.org"))
+    checkRouter(actionRequest, "actionWithParams")
+    verify(instance).actionWithParams(Some(ComplexEmailType("daphne", "coursera.org")))
+  }
+
+  @Test
+  def actionWithoutParamsTest(): Unit = {
+    val actionRequest = mkRequest(None, "POST", query = Map("action" -> "actionWithoutParams"))
+    checkRouter(actionRequest, "actionWithoutParams")
+    verify(instance).actionWithoutParams()
   }
 
   @Test
-  def parameterlessTest(): Unit = {
-    val actionRequest = mkRequest(None, "POST", query = Map("action" -> "parameterless"))
-    checkRouter(actionRequest, "parameterless")
-    verify(instance).parameterless()
+  def noCustomTypes(): Unit = {
+    checkNoCustomTypes("getAll")
+    checkNoCustomTypes("get")
+    checkNoCustomTypes("multiGet")
+    checkNoCustomTypes("create")
+    checkNoCustomTypes("update")
+    checkNoCustomTypes("delete")
+    checkNoCustomTypes("complex")
+    checkNoCustomTypes("actionWithoutParams")
```

Comment:
what about `actionWithParams`? 

Also, I think the reason some of these handlers have no custom types is because of `???` and others because they are not supposed to have custom types, maybe make that distinction clear? Or I could be missing the code completely, let me know!

Reply:
Auth type exists only if there is a `.auth` statement in the builder. For `get`, `multiGet` and `finder`, there is no notion of custom request or body type. For `create` and `update`, custom input body types are allowed, but only if explicitly declared with a`.jsonBody` statement. Same for `action`, with the addition that `action[ResponseType]` sets a custom response type.

So, nothing to do with the `???`: it's all about the `Nap` builders are called.